libc/test/src/string/CMakeLists.txt

```diff
@@ -374,7 +383,7 @@ add_libc_test(
     libc.src.string.memset_explicit
 )
 
-add_libc_multi_impl_test(memcmp libc-string-tests SRCS memcmp_test.cpp)
+add_libc_multi_impl_test(memcmp UNIT_TEST_ONLY libc-string-tests SRCS memcmp_test.cpp)
```

Comment:
This is the initial trial and it works, however I'm not aware of the use of hermetic tests. So a change here might cause regression that I'm not aware of yet. 
@michaelrj-google @nickdesaulniers